src/figuresClasses.ts

```diff
@@ -1,19 +1,87 @@
-export interface Figure {
+type Color = 'red' | 'green' | 'blue';
 
+enum FigureType {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
 }
 
-export class Triangle {
-
+export interface Figure {
+  color: Color;
+  shape: FigureType;
+  getArea(): number;
 }
 
-export class Circle {
+export class Triangle implements Figure {
+  shape = FigureType.Triangle;
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Wrong value for Triangle');
+    }
+
+    if (
+      a + b <= c
+      || a + c <= b
+      || b + c <= a
+    ) {
+      throw new Error('Wrong value for Triangle');
```

Comment:
What value is wrong? You can give the more descriptive message: `Impossible to build a triangle with given sides`